keras/src/models/functional.py

```diff
@@ -111,6 +116,9 @@ def __init__(self, inputs, outputs, name=None, **kwargs):
                         f"which has name '{v.name}'. Change the tensor name to "
                         f"'{k}' (via `Input(..., name='{k}')`)"
                     )
+        else:
+            self._input_names = None
+            self._inputs_struct = inputs
```

Comment:
This is already done by `Function.__init__` line 144. In fact `Function.__init__` will overwrite what you set here.